ddtrace/llmobs/decorators.py

```diff
@@ -66,8 +69,20 @@ def wrapper(*args, **kwargs):
                 if span_name is None:
                     span_name = func.__name__
                 traced_operation = getattr(LLMObs, operation_kind, "workflow")
-                with traced_operation(name=span_name, session_id=session_id, ml_app=ml_app):
-                    return func(*args, **kwargs)
+                with traced_operation(name=span_name, session_id=session_id, ml_app=ml_app) as span:
+                    func_signature = signature(func)
+                    bound_args = func_signature.bind_partial(*args, **kwargs)
+                    if _automatic_io_annotation and bound_args.arguments and span.get_tag(INPUT_VALUE) is None:
```

Comment:
nit: we might not need the `span.get_tag(INPUT_VALUE)` check here, right? i think the only case this could be set is if somehow `LLMObs.annotate()` was called before the function was invoked. but correct me if I'm wrong! i think how it was before already allowed users to override automatic annotations for inputs, it was just the outputs that needed the change.